core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpServer.java

```diff
@@ -0,0 +1,150 @@
+
+    @Nullable
+    public static LineTcpServer create(
+            CairoConfiguration cairoConfiguration,
+            LineTcpReceiverConfiguration lineConfiguration,
+            WorkerPool sharedWorkerPool,
+            Log log,
+            CairoEngine cairoEngine,
+            MessageBus messageBus
+    ) {
+        if (!lineConfiguration.isEnabled()) {
+            return null;
+        }
+
+        ServerFactory<LineTcpServer, WorkerPoolAwareConfiguration> factory = (netWorkerPoolConfiguration, engine, workerPool, local, bus,
+                functionfactory) -> new LineTcpServer(
+                        cairoConfiguration,
+                        lineConfiguration,
+                        cairoEngine,
+                        workerPool,
+                        bus);
+        LineTcpServer server = WorkerPoolAwareConfiguration.create(lineConfiguration.getWorkerPoolConfiguration(), sharedWorkerPool, log, cairoEngine, factory, messageBus, null);
+        return server;
+    }
+
+    private final IODispatcher<LineTcpConnectionContext> dispatcher;
+    private final LineTcpConnectionContextFactory contextFactory;
+    private final LineTcpMeasurementScheduler scheduler;
+    private final ObjList<LineTcpConnectionContext> busyContexts = new ObjList<>();
+
+    public LineTcpServer(
+            CairoConfiguration cairoConfiguration,
+            LineTcpReceiverConfiguration lineConfiguration,
+            CairoEngine engine,
+            WorkerPool workerPool,
+            MessageBus messageBus
+    ) {
+        this.contextFactory = new LineTcpConnectionContextFactory(engine, lineConfiguration, messageBus, workerPool.getWorkerCount());
+        this.dispatcher = IODispatchers.create(
+                lineConfiguration
+                        .getNetDispatcherConfiguration(),
+                contextFactory);
+        workerPool.assign(dispatcher);
+        scheduler = new LineTcpMeasurementScheduler(cairoConfiguration, lineConfiguration, engine, workerPool);
+        final IORequestProcessor<LineTcpConnectionContext> processor = (operation, context) -> {
+            if (context.handleIO()) {
+                busyContexts.add(context);
+            }
+        };
+        workerPool.assign(new SynchronizedJob() {
+            @Override
+            protected boolean runSerially() {
+                int n = busyContexts.size();
+                while (n > 0) {
+                    n--;
+                    if (!busyContexts.getQuick(n).handleIO()) {
+                        busyContexts.remove(n);
+                    }
+                }
+                return dispatcher.processIOQueue(processor);
+            }
+        });
+
+        for (int i = 0, n = workerPool.getWorkerCount(); i < n; i++) {
+            // http context factory has thread local pools
+            // therefore we need each thread to clean their thread locals individually
+            workerPool.assign(i, () -> {
```

Comment:
this isn't a big deal, but on a server with 20 cores there will be 20 identical classes created by java for this lambda. Perhaps we can create lambda instance outside of the loop?